docs/src/locators.md

```diff
@@ -188,121 +129,367 @@ await page.Locator("button").CountAsync();
-# Note: the code inside evaluateAll runs in page, you can call any DOM apis there.
-texts = rows.evaluate_all("list => list.map(element => element.textContent)")
+Most form controls usually have dedicated labels that could be conveniently used to interact with the form. Input actions in Playwright automatically distinguish between labels and controls, so you can just locate the label to perform an action on the associated control.
+
+For example, consider the following DOM structure.
+
+```html
+<label for="password">Password:</label><input type="password">
+```
+
+You can target the label with something like `text=Password` and perform the following actions on the password input:
+- `click` will click the label and automatically focus the input field;
+- `fill` will fill the input field;
+- `inputValue` will return the value of the input field;
+- `selectText` will select text in the input field;
+- `setInputFiles` will set files for the input field with `type=file`;
+- `selectOption` will select an option from the select box.
+
+```js
+// Fill the input by targeting the label.
+await page.locator('text=Password').fill('secret');
 ```
 
 ```java
-// Locate elements, this locator points to a list.
-Locator rows = page.locator("table tr");
+// Fill the input by targeting the label.
+page.locator("text=Password").fill("secret");
+```
 
-// Pattern 1: use locator methods to calculate text on the whole list.
-List<String> texts = rows.allTextContents();
+```python async
+# Fill the input by targeting the label.
+await page.locator('text=Password').fill('secret')
+```
 
-// Pattern 2: do something with each element in the list.
-int count = rows.count()
-for (int i = 0; i < count; ++i)
-  System.out.println(rows.nth(i).textContent());
+```python sync
+# Fill the input by targeting the label.
+page.locator('text=Password').fill('secret')
+```
 
-// Pattern 3: resolve locator to elements on page and map them to their text content.
-// Note: the code inside evaluateAll runs in page, you can call any DOM apis there.
-Object texts = rows.evaluateAll("list => list.map(element => element.textContent)");
+```csharp
+// Fill the input by targeting the label.
+await page.Locator("text=Password").FillAsync("secret");
+```
+
+However, other methods will target the label itself, for example `textContent` will return the text content of the label, not the input field.
+
+### Locate in a subtree
+
+You can chain [`method: Page.locator`] and [`method: Locator.locator`] calls to narrow down the search to a particular part of the page.
+
+For example, consider the following DOM structure:
+
+```html
+<div data-test-id='product-card'>
+  <span>Product 1</span>
+  <button>Buy</button>
+</div>
+<div data-test-id='product-card'>
+  <span>Product 2</span>
+  <button>Buy</button>
+</div>
+```
+
+You can now click a button for a particular product:
+
+```js
+// Find a product card that contains text "Product 2"
```

Comment:
i suggest adding this text outside of code block and just make it a sentence. Find a product....then....